src/runtime/GlobalObjectBuiltinRegExp.cpp

```diff
@@ -350,6 +350,139 @@ static Value builtinRegExpSplit(ExecutionState& state, Value thisValue, size_t a
     return A;
 }
 
+static Value builtinRegExpReplace(ExecutionState& state, Value thisValue, size_t argc, Value* argv, bool isNewExpression)
+{
+    ASSERT(argv != nullptr);
+    RESOLVE_THIS_BINDING_TO_OBJECT(obj, RegExp, replace);
+    RegExpObject* rx = obj->asRegExpObject(state);
+    String* str = argv[0].toString(state);
+    ASSERT(str != nullptr);
+    Value replaceValue = argv[1];
+    size_t lengthStr = str->length();
+    bool functionalReplace = replaceValue.isCallable();
+    bool fullUnicode = false;
+    String* replacement = String::emptyString;
+    // 7. If functionalReplace is false, then
+    if (functionalReplace) {
+        replaceValue = replaceValue.toString(state);
+    }
+
+    bool global = rx->option() & RegExpObject::Option::Global;
```

Comment:
All the named `RegExp flags` must be accessed with the `rx->get(....).value(...).toBoolean(...)` ways as the standard describes.